datacube_alchemist/_utils.py

```diff
@@ -66,11 +66,7 @@ def _write_stac(
 ):
     out_dataset = serialise.from_path(metadata_path)
     stac_path = Path(str(metadata_path).replace("odc-metadata.yaml", "stac-item.json"))
-    # Madness in deferred destination logic
-    uri_base = dataset_assembler.names.destination_folder(
-        Path(task.settings.output.location)
-    )
-    uri_base = str(uri_base) + "/"
+    uri_base = dataset_assembler.names.dataset_location
```

Comment:
the dataset_assembler.names.dataset_location == temp_dir. 

Could you check your output stac-json files? I believe we should put the task.settings.output.location in front of it? Or the stac-metadata will use the temp_dir as the path.

Reply:
You're right